### ISSUE TITLE
Auto format error

### ISSUE THREAD

**Issue by Johanvdberg** [eclipse/xtext-core#145]
Using the project created by with wizard with a custom grammar and using a file with context

module  
end_module

in my plug in. I get a list list of exception, it is not exactly the same of the post  [http://stackoverflow.com/questions/40009492/xtext-formatting-exception?noredirect=1#comment67331354_40009492](on stack overflow) but looks for me like the same error.

I get the error when I auto format, ctrl+shift+f

[auto_format_log.txt](https://github.com/eclipse/xtext-core/files/527863/auto_format_log.txt)

[auto_format_error.zip](https://github.com/eclipse/xtext-core/files/527858/auto_format_error.zip)

**Comment by meysholdt** [comment]
What Xtext version did you use? Did you use a version that includes this fix: 1cdd0ab042515a5ae8cf5caa11c200391a3e9db8 ?

**Comment by Johanvdberg** [comment]
I am not sure, how can I check?

I use:
Eclipse DSL Tools

Version: Neon.1a Release (4.6.1)
Build id: 20161007-1200

**Comment by spoenemann** [comment]
Help &rarr; Installation Details &rarr; Features &rarr; look for `org.eclipse.xtext.*`

**Comment by Johanvdberg** [comment]
org.eclipse.xtend.sdk (2.10.0.v201605250459) "Xtend IDE"
org.eclipse.xtext.sdk (2.10.0.v201605250459) "Xtext "
org.eclipse.xtext.ui (2.10.0.v201605250459) "Xtext UI Core "
org.eclipse.xtext.xbase (2.10.0.v201605250459) "Xbase Model "
org.eclipse.xtext.xbase.lib (2.10.0.v201605250459) "Xbase Runtime Library"
org.eclipse.xtext.xtext.ui (2.10.0.v201605250459) "Xtext Xtext UI "

**Comment by cdietrich** [comment]
can be reproduced with 2.11. i assume this is cause by having no AST at all in this case

changing the grammar to

```
Model:
	{Model}'module'
	(func=Data2 |
	test_func=TestFunction)?
	'end_module';
```

seems to solve this

**Comment by Johanvdberg** [comment]
I also get that adding {Model} result in calling of the format functions.

Why is {Model} needed?

**Comment by cdietrich** [comment]
if forces the model root to be created even is you have no assignments (which is not done by default, and which is why you get the warning on the .xtext file

**Comment by Johanvdberg** [comment]
This was marked as a bug, but my code now works. Must I still do something? (I am new to reporting bugs)

**Comment by svenefftinge** [comment]
No, you don't need to do something. We should just deal a bit better with empty resources in the formatter. The warning in the grammar is already there, so not a super big deal.

**Comment by spoenemann** [comment]
Is this fixed?

**Comment by cdietrich** [comment]
no its not
its just avoided

@meysholdt need to look at but has no time/mood

**Comment by meysholdt** [comment]
I've been quite busy recently. Will try to look at it soon. Sorry for the delay.

**Comment by cdietrich** [comment]
leave this one open to find a general solution
